tests/platform_tests/test_advanced_reboot.py

```diff
@@ -14,8 +15,16 @@
 
 
 def pytest_generate_tests(metafunc):
+    input_sad_cases = metafunc.config.getoption("sad_case_list")
+    input_sad_list = list()
+    for input_case in input_sad_cases.split(","):
+        input_case = input_case.strip()
+        if input_case.lower() not in SAD_CASE_LIST:
+            logging.warn("Unknown SAD case ({}) - skipping it.".format(input_case))
+            continue
+        input_sad_list.append(input_case.lower())
     if "sad_case_type" in metafunc.fixturenames:
-        sad_cases = SAD_CASE_LIST
+        sad_cases = input_sad_list
```

Comment:
Does there need to be a check if `input_sad_list` is empty for whatever reason?

Reply:
Not explicitly needed, but it is handled implicitly as part of `metafunc.parametrize`. When the empty param (empty `input_sad_list` here) is sent, the test is collected and then skipped:
`SKIPPED [1] platform_tests/test_advanced_reboot.py:81: got empty parameter set ['sad_case_type'], function test_warm_reboot_sad at /var/src/sonic-mgmt/tests/platform_tests/test_advanced_reboot.py:80`